python/TestHarness/TestHarness.py

```diff
@@ -687,6 +695,12 @@ def cleanup(self):
     if self.num_failed == 0:
       self.writeState(self.executable)
 
+    # Close the failed_tests file. Ignore if the file is already closed (ctl-c, keyboard interrupt)
+    try:
+      self.writeFailedTest.close()
```

Comment:
Ah, now I see. You are worried about closing the file twice? That won't happen in at least one of these cases because we are existing in the other case. That means this one will always work. I suggest that you simply test to see if the file is closed above and then close. try/catch isn't the correct solution here.